### PR TITLE
[react] Add optional props type parameter to ReactType

### PR DESCRIPTION
**Pull request by pelotom**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- \[N/A\] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- \[N/A\] Increase the version number in the header if appropriate.
- \[N/A\] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by DovydasNavickas — APPROVED** [review]
Non-breaking change. I guess, 👍

**Comment by RyanCavanaugh** [comment]
Approved by a listed owner. PR ready to merge pending express review by a maintainer.

**Comment by johnnyreilly** [comment]
Why not?

**Pull request merged**
